src/gstreamer/gstkvssink.cpp

```diff
@@ -318,12 +318,15 @@ void kinesis_video_producer_init(GstKvsSink *kvssink)
         session_token_str = string(kvssink->session_token);
     }
 
-    if (nullptr == (default_region = getenv(DEFAULT_REGION_ENV_VAR))) {
-        region_str = string(kvssink->aws_region);
+    if(IS_EMPTY_STRING(kvssink->aws_region)) {
```

Comment:
```suggestion
    if (IS_EMPTY_STRING(kvssink->aws_region)) {
```

Also, let's use the GETENV macro from PIC.

For LOG_AND_THROW, can we have the string dynamically created based on the the env var and property names, and also include the fallback order if both are set in the message?

Reply:
Fixed. For property name, there doesnt seem to be a direct way to get the name other than running a for loop and comparing properties which seems wasteful. So leaving it as is.

Reply:
I'm thinking we can extract `aws-region` string out as a macro

https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-cpp/blob/de4299f942bb89c867d0dc73aeb4bc875d560a8f/src/gstreamer/gstkvssink.cpp#L612

Reply:
Let's also update the description "AWS Region" to include the env var (other option).

Reply:
I think AWS region is ok. Adding env var is not relevant in the description of this property. Using a macro is fine, but I would like it to be consistent for all properties, which is beyond the scope of this PR.